.azure/pipelines/ci.yml

```diff
@@ -749,56 +749,13 @@ stages:
           includeForks: true
 
   # Source build
-  - job: Source_Build
-    displayName: 'Test: Linux Source Build'
-    container: centos:7
-    pool:
-      vmImage: 'ubuntu-18.04'
-    variables:
-      LC_ALL: 'en_US.UTF-8'
-      LANG: 'en_US.UTF-8'
-      LANGUAGE: 'en_US.UTF-8'
-      DotNetCoreSdkDir: $(Agent.ToolsDirectory)/dotnet
-      # This isn't needed in the path because build does not need to _use_ global tools.
-      DOTNET_CLI_HOME: $(System.DefaultWorkingDirectory)
-      DOTNET_SYSTEM_GLOBALIZATION_INVARIANT: true
-    steps:
-    - script: |
-        source eng/common/native/common-library.sh
-        mkdir -p $(System.DefaultWorkingDirectory)/.tools
-        GetFile https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 $(System.DefaultWorkingDirectory)/.tools/jq
-        chmod +x $(System.DefaultWorkingDirectory)/.tools/jq
-        echo "##vso[task.prependpath]$(System.DefaultWorkingDirectory)/.tools"
-      displayName: Install jq
-    - ${{ if ne(variables['System.TeamProject'], 'public') }}:
-        - task: Bash@3
-          displayName: Setup Private Feeds Credentials
-          inputs:
-            filePath: $(Build.SourcesDirectory)/eng/common/SetupNugetSources.sh
-            arguments: $(Build.SourcesDirectory)/NuGet.config $Token
-          env:
-            Token: $(dn-bot-dnceng-artifact-feeds-rw)
-    - script: ./eng/scripts/ci-source-build.sh --ci --nobl --configuration Release /p:BuildManaged=true /p:BuildNodeJs=false $(_InternalRuntimeDownloadArgs)
-      displayName: Run ci-source-build.sh
-    - task: PublishBuildArtifacts@1
-      displayName: Upload logs
-      condition: always()
-      continueOnError: true
-      inputs:
-        pathtoPublish: artifacts/log/
-        artifactName: Source_Build_Logs
-        artifactType: Container
-        parallel: true
-    - task: PublishBuildArtifacts@1
-      displayName: Upload package artifacts
-      # Only capture source build artifacts in PRs for the sake of inspecting
-      # changes that impact source-build. The artifacts from this build pipeline are never actually used.
-      condition: and(succeeded(), in(variables['Build.Reason'], 'PullRequest'))
-      inputs:
-        pathtoPublish: artifacts/packages/
-        artifactName: Source_Build_Packages
-        artifactType: Container
-        parallel: true
+  - template: /eng/common/templates/job/source-build.yml
+    parameters:
+      platform:
+        name: 'Managed'
+        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-f39df28-20191023143754'
+        buildScript: './eng/scripts/prepare-sourcebuild-globaljson.sh && ./eng/build.sh'
```

Comment:
Is it possible to invoke this within build.sh based on a source-build trigger - e.g. `/p:ArcadeBuildFromSource` or explicit source-build option?  This would provide a better local dev workflow as invoking `build.sh -c Release --restore --build --pack /p:ArcadeBuildFromSource=true` is a constant across all repos.

Reply:
Alternatively this could be invoked via a target after the inner sourcebuild clone.

Reply:
I'll apply it in the target.